### PR TITLE
content 16

### PR DESCRIPTION
**Pull request by mehmetcanozen**
Made with reference to "Latest News" section of this page:                                                                                                                                https://themes.pixelwars.org/archy/demo-01/

### CODE REVIEW

**Review by Ayazelar — COMMENTED** [review]
Image "alt" property shouldn't be an empty string, it can be title or subTitle. All HTML code should be SEO optimized.

File: editor-components/content/content16/content16.tsx [hunk 1]
```diff
@@ -1,133 +1,268 @@
...
+      displayer: "Items",
+      value: [
+        {
+          type: "object",
+          key: "slidercontent",
+          displayer: "Slider Content",
+          value: [
+            {
+              type: "string",
+              key: "image_subtitle",
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:49]: Use camelCase for all keys.

File: editor-components/content/content16/content16.tsx [hunk 2]
```diff
@@ -1,133 +1,268 @@
+import Slider from "react-slick";
 import * as React from "react";
 import { BaseContent } from "../../EditorComponent";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
 import styles from "./content16.module.scss";
-import ComposerLink from "../../../../custom-hooks/composer-base-components/Link/link";
+
+type Card = {
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:8]: Wrong format.

[Please read this.](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md)

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
Please handle the changes.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
try to make that animation without using keyframe. try to implement it with transition

**Comment by mehmetcanozen** [comment]
I could not switch the animations from keyframes to transition so I removed them altogether. Other than that made the requested changes.

File: editor-components/content/content16/content16.module.scss [hunk 3]
```diff
@@ -1,105 +1,303 @@
 .container {
-    width: 100%;
-    .max-content {
-        max-width: var(--composer-content-width);
+  width: 100%;
+  padding: 30px;
+
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.module.scss:4]: i guess .container needs background-color: var(--composer-html-background)

File: editor-components/content/content16/content16.tsx [hunk 4]
```diff
@@ -1,129 +1,325 @@
...
+type CardType = {
+  title?: string;
+  subtitle?: string;
+  description?: string;
+  image?: string;
+  imageSubtitle?: string;
+  imageTitle?: string;
+  imageDescription?: string;
+  url?: string;
+};
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:17]: why all optional? remove "?" from these.

also the types are wrong.

```javascript
title: JSX.Element;
subtitle: JSX.Element;
description: JSX.Element;
imageTitle: JSX.Element;
imageSubtitle: JSX.Element;
imageTitle: JSX.Element;
```

others can stay as the same

File: editor-components/content/content16/content16.tsx [hunk 5]
```diff
@@ -1,129 +1,325 @@
...
+        );
+      },
+    };
+    
+    const title = this.getPropValue("title");
+    const subtitle = this.getPropValue("subtitle");
+    const isTitleExist = this.castToString(title);
+    const isSubTitleExist = this.castToString(subtitle);
+    const items = this.castToObject<CardType[]>("items").filter(
+      (item: CardType) => item.image || item.imageTitle || item.imageSubtitle || item.url
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:260]: imageTitle and imageSubtitle should be casted to string

File: editor-components/content/content16/content16.tsx [hunk 6]
```diff
@@ -1,129 +1,325 @@
...
+            </header>
+          )}
+          <main className={this.decorateCSS("wrapper")}>
+            {items.length > 0 && (
+              <div className={this.decorateCSS("slider-parent")}>
+                <Slider {...settings} className={this.decorateCSS("carousel")}>
+                  {items.map((item: CardType, index: number) => (
+                    <article
+                      className={`${this.decorateCSS("slider-inner-div")} ${this.getComponentState("prevSlide") === index ? this.decorateCSS("prevSlide") : ""} ${this.getComponentState("nextSlide") === index ? this.decorateCSS("nextSlide") : ""}`}
+                      key={`sld-16-${index}`}
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:288]: you can just say `key={index}`

File: editor-components/content/content16/content16.tsx [hunk 7]
```diff
@@ -1,129 +1,325 @@
...
+                    >
+                      <div className={this.decorateCSS("content-div")}>
+                        {item.image && (
+                          <div className={this.decorateCSS("img-div")}>
+                            <a href={item.url} target="_blank" rel="noopener noreferrer">
+                              <img
+                                alt={item.imageTitle || item.imageSubtitle}
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:295]: they should be casted to string

File: editor-components/content/content16/content16.tsx [hunk 8]
```diff
@@ -1,129 +1,325 @@
...
+                                {item.imageSubtitle}
+                              </h3>
+                            )}
+                            {item.imageTitle && (
+                              <h2 className={this.decorateCSS("item-title")}>
+                                {item.imageTitle}
+                              </h2>
+                            )}
+                          </div>
+                        )}
```
Comment (dogusmiuw) [inline editor-components/content/content16/content16.tsx:315]: they should be casted to string while doing empty check, because they return JSX element.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

**Comment by dogusmiuw** [comment]
exclude calltoaction7 files from this pr

File: editor-components/content/content16/content16.tsx [hunk 9]
```diff
@@ -1,129 +1,624 @@
+import Slider from "react-slick";
```
Comment (Ayazelar) [inline editor-components/content/content16/content16.tsx:1]: use composerslider

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/content/content16/content16.module.scss [hunk 10]
```diff
@@ -1,105 +1,384 @@
...
+@keyframes underlineExpand {
+  from {
+    width: 0;
+  }
+  to {
     width: 100%;
-    .max-content {
-        max-width: var(--composer-content-width);
+  }
+}
```
Comment (Ayazelar) [inline editor-components/content/content16/content16.module.scss:8]: remove

File: editor-components/content/content16/content16.tsx [hunk 11]
```diff
@@ -1,133 +1,496 @@
...
+                            : ""
+                          }`}
+                        key={index}
                       >
-                        <button
-                          className={`${this.decorateCSS("button")}`}
+                        <div
+                          className={this.decorateCSS("content-div")}
+                          style={
+                            this.castToString(item.imageTitle) ? {} : { padding: "20px 10px" }
```
Comment (Ayazelar) [inline editor-components/content/content16/content16.tsx:428]: remove inline style

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

**Pull request merged**